android/app/src/main/java/com/sample/react/TestAppReactNativeHost.kt

```diff
@@ -43,7 +56,11 @@ class TestAppReactNativeHost @Inject constructor(
                 .setJSMainModulePath(jsMainModuleName)
                 .addPackages(packages)
                 .setUseDeveloperSupport(useDeveloperSupport && !useEmbeddedBundle)
-                .setInitialLifecycleState(LifecycleState.RESUMED)
+                .setInitialLifecycleState(if (currentActivity == null) {
+                    LifecycleState.BEFORE_CREATE
```

Comment:
This is ambiguous, as we don't know if activity going to be created at all if it is null. I can accept it as long as we're going to refactor this to rely on the `ReactActivityDelegate` to manage the `currentActivity` instance later.

Reply:
Isn't that fine though? Why would RN care whether the Activity gets created if the initial state is correct?

Reply:
I checked the source code of the `ReactNativeHost`, and they make the same assumption on the activity state. My concern was that you can set the state to `BEFORE_CREATE`, but then activity is actually never created and you end-up in some weird state.